ElectronClient/gui/ConfigScreen.jsx

```diff
@@ -270,8 +280,22 @@ class ConfigScreenComponent extends React.Component {
 			borderRadius: 4,
 		});
 
+		const detectTheme = (value) => {
+			this.setState({ shouldUseDarkColors: bridge().shouldUseDarkColors() });
+
+			// active theme should change to preferred theme based on system detected theme
+			if (value == true) {
+				this.state.shouldUseDarkColors ? updateSettingValue('theme', settings['preferredDarkTheme']) : updateSettingValue('theme', settings['preferredLightTheme']);
+			} else {
+				updateSettingValue('theme', settings['theme']);
+			}
```

Comment:
Perhaps this function would be clearer if you could refactor it to have only one `updateSettingValue` call at the bottom.

Reply:
Okay sure. I'll try!